velox/exec/tests/PrestoQueryRunnerTimeTransformTest.cpp

```diff
@@ -0,0 +1,119 @@
+#include "velox/exec/tests/PrestoQueryRunnerIntermediateTypeTransformTestBase.h"
+
+namespace facebook::velox::exec::test {
+namespace {
+
+class PrestoQueryRunnerTimeTransformTest
+    : public PrestoQueryRunnerIntermediateTypeTransformTestBase {
+ public:
+  VectorPtr createTimeVector(const vector_size_t size) {
+    std::vector<std::optional<int64_t>> values;
+    values.reserve(size);
+
+    for (vector_size_t i = 0; i < size; ++i) {
+      if (i % 10 == 0) {
+        values.push_back(std::nullopt); // Some nulls for testing
+      } else {
+        // Create realistic TIME values: 0 to 86399999 (23:59:59.999)
+        values.push_back((i * 3661000 + 123) % 86400000);
+      }
+    }
+
+    return makeNullableFlatVector(values, TIME());
+  }
+};
+
+// Test that TIME is recognized as an intermediate type that needs
+// transformation
+TEST_F(PrestoQueryRunnerTimeTransformTest, isIntermediateOnlyType) {
+  // Core test: TIME should be an intermediate type
+  ASSERT_TRUE(isIntermediateOnlyType(TIME()));
+
+  // Complex types containing TIME should also be intermediate types
+  ASSERT_TRUE(isIntermediateOnlyType(ARRAY(TIME())));
+  ASSERT_TRUE(isIntermediateOnlyType(MAP(TIME(), BIGINT())));
+  ASSERT_TRUE(isIntermediateOnlyType(MAP(BIGINT(), TIME())));
+  ASSERT_TRUE(isIntermediateOnlyType(ROW({TIME(), BIGINT()})));
+}
+
+// Test that TIME vectors can be transformed for fuzzer persistence/testing
+TEST_F(PrestoQueryRunnerTimeTransformTest, transformTimeVector) {
+  // Core test: verify TIME vector can be transformed (TIME -> BIGINT)
+  auto timeVector = createTimeVector(100);
+  test(timeVector); // This tests the TIME->BIGINT transform works
+}
+
+// Test TIME with all encodings: flat, dictionary, and constant
+// Dictionary encodings: 3 variants (no nulls, some nulls, all nulls)
+// Constant encodings: 2 variants (non-null constant, null constant)
+TEST_F(PrestoQueryRunnerTimeTransformTest, transformTimeEncodings) {
+  test(TIME());
+}
+
+// Test TIME in arrays with dictionary and constant encodings
+// Dictionary encoded arrays: wraps array in dict with various null patterns
+// Constant encoded arrays: wraps array as constant vector
+TEST_F(PrestoQueryRunnerTimeTransformTest, transformTimeArray) {
+  auto timeVector = createTimeVector(100);
+  testArray(timeVector); // internally calls testDictionary() and testConstant()
+}
+
+// Test TIME in maps with dictionary and constant encodings
+// Dictionary encoded maps: wraps map in dict with various null patterns
+// Constant encoded maps: wraps map as constant vector
+TEST_F(PrestoQueryRunnerTimeTransformTest, transformTimeMap) {
+  // create TIME keys without nulls since map keys can't be null
+  std::vector<std::optional<int64_t>> keyValues;
+  keyValues.reserve(100);
+  for (vector_size_t i = 0; i < 100; ++i) {
+    keyValues.push_back((i * 3661000 + 123) % 86400000);
+  }
+  auto keys = makeNullableFlatVector(keyValues, TIME());
+
+  auto values = makeNullableFlatVector(
```

Comment:
Any reason for manually writing up this Value vector - why not have it being generated just like keyValues ?